pylabnet/hardware/staticline/staticline.py

```diff
@@ -0,0 +1,153 @@
+from pylabnet.utils.logging.logger import LogHandler
+
+
+class StaticLine():
+
+    def __init__(self, name, logger,  hardware_module,  **kwargs):
+        '''High level staticline class.
+
+        This class is used in conjunction with hardware modules to send out digital
+        signals ('voltage low' and 'voltage high'). This top level class is hardware agnostic.
+        With the use of a StaticLineHardwareHandler, this class will be associated with the necessary
+        setup functions and output functions of a hardware module.
+
+        :name: A easily recognizable name for this staticline, ideally referring to the device being controlled
+            by it, e.g. 'Shutter 1'.
+        :logger: An instance of a LogClient.
+        :hardware_module: An instance of a hardware module. Based on this, the corresponsind setup function is automatically
+            called.
+        '''
+
+        self.name = name
+
+        # Instantiate log
+        self.log = LogHandler(logger=logger)
+
+        # Instanciate Hardware_handler
+        self.hardware_handler = StaticLineHardwareHandler(hardware_module, self.log, name, **kwargs)
+
+    def up(self):
+        '''Set output to high.'''
+        self.hardware_handler.up()
+        self.log.info(
+            f"Staticline {self.name} set to high."
+        )
+
+    def down(self):
+        '''Set output to low.'''
+        self.hardware_handler.down()
+        self.log.info(
+            f"Staticline {self.name} set to low."
+        )
+
+
+class StaticLineHardwareHandler():
```

Comment:
If I understand correctly, you would have to edit this class in order to use different hardware (e.g. NI card) for static line DIO?

Reply:
Exactly. You would write a hardware-specific setup member function of `StaticLineHardwareHandler`, say `_setup_NIDAQDriver`, and register it by adding an entry to the following dict:

```python
registered_staticline_modules = {
     'HDAWGDriver':  self._setup_HDWAGDriver,
     'NIDAQDriver':  self._setup_NIDAQDriver
}
```